homeworks/veizerattila/3_conditionals_loops/3_rock_paper_scissors.py

```diff
@@ -0,0 +1,75 @@
+﻿"""Feladat 2: while és for loop használata
+- Hozz létre egy rock_paper_scissors.py nevű file-t, és kódold le a következő feladat megoldását:
+- Írj egy python programot ami levezényli a kő-papír-olló játékot két játékos között.
+    > A program kérje be, hogy hány kört akarnak játszani a játékosok.
+    > Figyelj oda, hogy olyan számot kell megadnia a felhasználónak ami mellett nem tudnak döntetlent játszani! Ha nem ilyen számot ad meg, írj ki hibaüzenetet és addig kérd be újra a körök számát amíg páratlan számot nem ad meg
+    > Ezután a program felváltva kérje be az első és második játékos válaszát, ami kizárólag a következő stringek valamelyik lehet: "rock", "paper", "scissors". Ellenkező esetben kezeld úgy a hibát ahogy a körök számánál.
+    >Tárold a nyertesek pontjait, és minden kör végén növeld az aktuális játékos pontszámát.
+    > A végén printeld ki ki nyert, és hány ponttal.
+"""
+
+# Változók deklarálása, ill. azon inpputok deklarálása, ami már a futás elején kelleni fog:
+print("Let's play 'Rock-Paper-Scissors' with 2 players")
+input("Press Enter if you're ready!")
+actions = ['Rock', 'Paper', 'Scissors']
+player1 = input("Name of Player1: ").strip().title()
+player2 = input("Name of Player2: ").strip().title()
+feedback = "Good" # köztes visszajelzés, az egyes input értékek sikeres elfogadásakor
+
+# Először két külön változóban definiáltam a két játékos játékát, de itt elég összetett, így egy function-ba gyúrtam:
+def valid_gamer_input(player, action):  # egyedi function definiálása, a későbbi játéklépések támogatására
+    while True:
+        action = input(f"{player}: What's your game? ").title()
+        if action not in actions:
+            print (f"{action} is an invalid input from {player}. Only {actions} are accepted. Try again. ")
+        else:
+            print(feedback)
+            return action
+
+# A külső while ciklus az újrajátszásra kérdez rá, a játék végén:
+while True:
+    points1 = 0
+    points2 = 0
+    # Belső while ciklus a körök bekérésére:
+    while True: 
+        rounds = int(input("How many rounds do you want to play? "))
+        if rounds < 3 or rounds % 2 == 0 :
+            print(f"{rounds} rounds won't declare a clear winner. Number of rounds must be odd. Please start over.")
+        else:
+            print(feedback)
+            break    
+
+    # Beágyazott while ciklus a játék futtatására:
+    a = 0
+    while a < (rounds):
+        # itt hozom be a fent definiált function-t:
+        action1 = valid_gamer_input(player1, actions)
+        action2 = valid_gamer_input(player2, actions)
+
+        rounds -= 1
+
+        # játszma logikák definiálása és pontok számlálása
+        if action1 == action2:
+            print(f"Both {player1} and {player2} selected {action1}, it's a draw. No point gained.")
```

Comment:
ide betehetnél egy` continue`-t, és a 49. soron lévő rounds -= 1-et áttehetnéd a 60. sorra.
Így elérnéd azt, hogy a continue-val újrakezdődjön az adott kör, ha ugyanazt az értéket adják meg (így ugye a continue miatt nem ér el a rounds -= 1-ig. Ezáltla se egy adott körben, se a játék végeredményeként nem születhet döntetlen.
Így nem kell a 69. soron az else ág, és a játékmenet is tükrözi a valóságot.

Reply:
Szia István, köszi szépen a jelzést. Pontosítottam a kódót, ahogy javasoltad. Így valóban lekezeli a döntetlent is úgy, hogy azt nem számolja bele a körökbe.

Emellett a points1 és point2 változót is behoztam a while alá, hogy új játék indításakor lenullázódjon a pontok száma is, és ne pedig folytassa a számlálást.